simplecpp.cpp

```diff
@@ -2341,6 +2341,63 @@ static void simplifySizeof(simplecpp::TokenList &expr, const std::map<std::strin
     }
 }
 
+/** Evaluate __has_include(file) */
+static std::string openHeader(std::ifstream &f, const simplecpp::DUI &dui, const std::string &sourcefile, const std::string &header, bool systemheader);
+static void simplifyHasInclude(simplecpp::TokenList &expr, const simplecpp::DUI &dui)
+{
+    for (simplecpp::Token *tok = expr.front(); tok; tok = tok->next) {
+        if (tok->str() != "__has_include")
+            continue;
+        simplecpp::Token *tok1 = tok->next;
+        if (!tok1) {
+            throw std::runtime_error("missing __has_include argument");
+        }
+        simplecpp::Token *tok2 = tok1->next;
```

Comment:
do we need to handle comments? is this valid code:

    __has_include /*comment*/ ("foo.h")

Reply:
This is valid code, and my testing (https://github.com/al42and/simplecpp/commit/64929e8775dc12011460fbb47795f0b1fea159ea) shows that it's processed correctly.